_studio/shared/src/libmfx_core_vaapi.cpp

```diff
@@ -630,7 +630,7 @@ VAAPIVideoCORE::AllocFrames(
                 // Checking for unsupported mode - external allocator exist but Device handle doesn't set
                 MFX_CHECK(m_Display, MFX_ERR_UNSUPPORTED)
 
-                if (response->NumFrameActual < request->NumFrameMin)
+                if (response->NumFrameActual && (!response->mids || (response->NumFrameActual < request->NumFrameMin)))
```

Comment:
This PR seems to be only for vaapi buffer? May it need for system memory buffer?

Reply:
yes, I only tested vaapi, but for system memory buffer. It may not need this. since it will copy the buffer.

Reply:
the code is simplified, now we do not need this line anymore.
@wangyan-intel please help review it again.
thanks